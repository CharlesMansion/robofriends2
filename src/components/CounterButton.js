import React, {Component} from 'react';

class CounterButton extends Component {
  constructor(props) {
    super(props);
    this.state = {
      count: 0
    };
  }

  shouldComponentUpdate(nextProps, nextState) {
    if (this.state.count !== nextState.count) {
      return true;
    }
    return false;
  }

  render() {
    return (
      <button 
        className='ma3 pa2 f4 black bg-navy yellow shadow-5 dib br3'
        id='counter'
        color={this.props.color}
        onClick={() => this.setState(state => ({count: state.count + 1}))}>
        Robot-o-meter : {this.state.count} 
      </button>
    );
  }
}

export default CounterButton