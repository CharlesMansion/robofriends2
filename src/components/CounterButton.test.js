import React from 'react';
import {shallow} from 'enzyme'
import CounterButton from './CounterButton';

describe('testing statefulness of CounterButton', () => {
	const wrapper = shallow(<CounterButton />)
	const instance = wrapper.instance();

	it('renders correctly', () => {
		expect(wrapper).toMatchSnapshot();
		wrapper.find('[id="counter"]').simulate('click');
		expect(instance.state.count).toEqual(1);
	})
})

