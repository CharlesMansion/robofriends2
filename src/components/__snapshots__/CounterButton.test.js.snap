// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing statefulness of CounterButton renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CounterButton />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Robot-o-meter : ",
        0,
      ],
      "className": "ma3 pa2 f4 black bg-navy yellow grow shadow-5 dib br3",
      "color": undefined,
      "id": "counter",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      "Robot-o-meter : ",
      0,
    ],
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Robot-o-meter : ",
          0,
        ],
        "className": "ma3 pa2 f4 black bg-navy yellow grow shadow-5 dib br3",
        "color": undefined,
        "id": "counter",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        "Robot-o-meter : ",
        0,
      ],
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
